src/parcsr_ls/par_2s_interp_device.c

```diff
@@ -62,11 +61,6 @@ hypre_BoomerAMGBuildModPartialExtInterpDevice( hypre_ParCSRMatrix  *A,
    HYPRE_Int          *Soc_diag_j   = hypre_ParCSRMatrixSocDiagJ(S);
    HYPRE_Int          *Soc_offd_j   = hypre_ParCSRMatrixSocOffdJ(S);
 
-   CF_marker_dev = hypre_TAlloc(HYPRE_Int, A_nr_local, HYPRE_MEMORY_DEVICE);
```

Comment:
@waynemitchell I think we missed the change here. With the IntArray, `CF_marker` should be on the device now. Does it look good to you?

Reply:
Yes, I agree that this extra copy to the device should be unnecessary now. Good catch!

Reply:
@waynemitchell Thanks! I am still puzzled why the following copy even worked. We were asking an H2D copy but the memory is actually from device. Any ideas?

Reply:
Oh... that does seem suspicious. I'm not sure. Where all has this been tested? I remember you mentioning that lassen has a unique treatment of memory that allows for some bad behavior as far as accessing memory between host and device. I don't know whether that has anything to do with it.

Reply:
@waynemitchell Yes, it has been tested in the regression. It may be because lassen can access host memory from device. But it doesn't explain the test on ray... Anyways, the copy is gone now :)